### DISCUSSION TITLE
How do you pass client values to client.Go?

### DISCUSSION THREAD

**Discussion by SteveMaybe** [qlova/seed#15] [Q&A] [answered]
Sorry for all the questions 😬, and thanks for bearing with me! Once I have a better understanding of how this all comes together, hopefully I can help contribute some examples. 

I'm trying to understand how to pass the clientside variables back to the server. 

```go
func main() {
	dataInput := &clientside.String{}

	app.New("Hello World",
		div.New(
			text.Set("What is your name? "),
			textbox.New(textbox.Update(dataInput), client.OnChange(client.Go(func() {
				fmt.Printf("Their name is %s\n", dataInput.GetString())
			}))),
		),
	).Launch()
}
````

**Comment by SteveMaybe** [comment, marked as answer]
Aha, I got it!

```go
func main() {
	dataInput := &clientside.String{}

	app.New("Hello World",
		div.New(
			text.Set("What is your name? "),
			textbox.New(textbox.Update(dataInput), client.OnClick(client.Go(func(a string) {
				fmt.Printf("Their name is %s\n", a)
			}, dataInput))),
		),
	).Launch()
}
```

**Reply by Splizard** [reply]
Yes, that's it! I would appreciate any examples you would like to contribute I think this Q&A section is also a good place for learning how to use the module. It's like a mini StackOverflow!
